src/scripts/main.js

```diff
@@ -1,3 +1,280 @@
 'use strict';
 
-// write code here
+// #region sorting Table
+const tableNames = document.querySelectorAll('thead th');
+let lastSortedIndex = null;
+const sortDirections = Array.from({ length: tableNames.length }, () => true);
+
+tableNames.forEach((element, index) => {
+  element.addEventListener('click', () => sortTable(index));
+});
+
+function sortTable(index) {
+  const tableBody = document.querySelector('tbody');
+  const headRows = Array.from(tableBody.querySelectorAll('tr'));
```

Comment:
The variable `headRows` is intended to store table rows, but the name `headRows` might be misleading since it suggests header rows. Consider renaming it to `bodyRows` for clarity.